Release UI - expandable history panel

bartaz opened the pull request:
Fixes #1267 

Opens releases history in expaning panel inside releases table and updates the styles.

#### DONE:

- [x] open history in expanding panel
- [x] update styles for 'tabs'
- [x] rename things (get rid of overlay related stuff)
- [x] open/close on cell click
- [x] clean up naming of css classes
- [x] remove arch column (when arch is filtered)
- [x] heading with `arch – channel`
- [x] hide indent of panel on smaller screens
- [x] don't indent first column without checkboxes

### QA

- ./run or https://snapcraft-io-canonical-websites-pr-1315.run.demo.haus/
- open Releases page of any snap
- click on any channel/arch cell to see history of releases
- click on 'unassigned revisions' to see all available revisions for given arch

<img width="1025" alt="screen shot 2018-11-15 at 17 00 40" src="https://user-images.githubusercontent.com/83575/48565113-0c423880-e8f8-11e8-8918-c47430592f9b.png">

webteam-app commented:
Starting demo at: https://snapcraft-io-canonical-websites-pr-1315.run.demo.haus/

webteam-app commented:
Starting demo at: https://snapcraft-io-canonical-websites-pr-1315.run.demo.haus/

therealjuan commented:
This looks quite good now. Just some small comments let me know what makes sense for this PR and what doesn't:

[ ] Reduce width of `Release date` column 
[ ] Have the same click to close functionality on `Show available revisions (...)` and `Release history:...`
[ ] In any channel history table, we display 10 revisions and a `Show more releases...` link. After clicking on the link it will display the remaining releases
[ ] We might want to hide releases already on a channel in any `Unassigned revisions`

This probably should be part of a new issue on github:

- To highlight more the items you are about to promote and perhaps only show the close button while on hover. At the moment the visual treatment is quite light
![image](https://user-images.githubusercontent.com/8167677/48611814-59b9b680-e987-11e8-8fd6-9e59afe6d798.png)
- Once you are about to promote a revision into a channel, that should be shown at the top of the table as a highlighted item

@lyubomir-popov wdyt?

bartaz commented:
@therealjuan 

> - [x] Reduce width of `Release date` column 
> - [x] Have the same click to close functionality on `Show available revisions (...)` and `Release history:...`

Both done.

> - [ ] In any channel history table, we display 10 revisions and a `Show more releases...` link. After clicking on the link it will display the remaining releases

It's a bigger piece of work, so I'd like it to be planned and implemented separately. Especially that if we implement this it will naturally lead users to expect us to load whole history from the store (that we don't do ATM).

> - [ ] We might want to hide releases already on a channel in any `Unassigned revisions`

As a quick workaround I renamed 'Unassigned revisions' to 'Available revisions'. If we want unassigned to only contain unreleased revisions we should create separate issue for that.

> - To highlight more the items you are about to promote and perhaps only show the close button while on hover. At the moment the visual treatment is quite light

As already designed a 'pending release' cells will have 'yellowish' colour. This is part of next visual updates task #1308 

> - Once you are about to promote a revision into a channel, that should be shown at the top of the table as a highlighted item

This I'd like to have addressed separately.

lyubomir-popov commented:
@bartaz well done. no comments aside for what @therealjuan has already mentioned. giving a plus one.

therealjuan commented:
Amazing! Just one small comment:

Can we update the copy of `Show available revisions(...)` to `Show all revisions(...)` it is confusing having 2 Available revisions on the same interface.

The unassigned revisions is reported here [#1327](https://github.com/canonical-websites/snapcraft.io/issues/1327)

therealjuan commented:
Just a few comments:

- Update the title to be `Latest revisions`
- Can we make the width of the architecture column to be `120px`?

Lukewh commented:
:hugs:

therealjuan commented:
LGTM!

Lukewh commented on static/js/publisher/release/releasesTable.js:
```diff
@@ -214,104 +220,152 @@ export default class RevisionsTable extends Component {
-        canBeClosed = false;
-        canBePromoted = false;
-      }
+    if (this.props.pendingCloses.includes(channel)) {
```
same here `||`'s

bartaz commented on static/js/publisher/release/releasesTable.js:
```diff
@@ -214,104 +220,152 @@ export default class RevisionsTable extends Component {
-      if (risk === UNASSIGNED) {
-        canBeClosed = false;
-      }
+    if (risk === UNASSIGNED) {
```
These 2 are not the same (one sets `canBePromoted` other `canBeClosed`).
